ui/apps/platform/src/Containers/Violations/Details/DeploymentDetails.tsx

```diff
@@ -70,41 +72,40 @@ const compareNetworkPolicies = (a: NetworkPolicy, b: NetworkPolicy): number => {
     return a.name.localeCompare(b.name);
 };
 
-const DeploymentDetails = ({ deployment }) => {
-    // attempt to fetch related deployment to selected alert
-    const [relatedDeployment, setRelatedDeployment] = useState(deployment);
-    const [namespacePolicies, setNamespacePolicies] = useState([]);
+export type DeploymentDetailsProps = {
+    alertDeployment: NonNullable<AlertViolation['deployment']>;
+};
 
-    useEffect(() => {
-        fetchDeployment(deployment.id).then(
-            (dep) => setRelatedDeployment(dep),
-            () => setRelatedDeployment(null)
-        );
-    }, [deployment.id, setRelatedDeployment]);
+const DeploymentDetails = ({ alertDeployment }: DeploymentDetailsProps) => {
+    const [namespacePolicies, setNamespacePolicies] = useState<NetworkPolicy[]>([]);
+
+    // attempt to fetch related deployment to selected alert
+    const { deployment: relatedDeployment, error: relatedDeploymentFetchError } =
+        useFetchDeployment(alertDeployment.id);
 
     useEffect(() => {
-        fetchNetworkPoliciesInNamespace(deployment.clusterId, deployment.namespace).then(
-            (policies) => setNamespacePolicies(policies.response.networkPolicies),
+        fetchNetworkPoliciesInNamespace(alertDeployment.clusterId, alertDeployment.namespace).then(
+            // TODO Infer type from response once NetworkService.js is typed
+            (policies: NetworkPolicy[]) => setNamespacePolicies(policies),
             () => setNamespacePolicies([])
         );
-    }, [deployment.namespace, deployment.clusterId, setNamespacePolicies]);
-
-    const deploymentObj = relatedDeployment || deployment;
```

Comment:
There are two deployments in use here, the deployment attached to the `Alert` object with minimal properties, and the `Deployment` received directly from the DeploymentService that has verbose details.

Instead of using a single variable to hold both, we use a clear separation to avoid issues with missing properties, ambiguous `undefined` objects, etc. when the deployment is needed in child components.

Reply:
Super!